dspace-server-webapp/src/test/java/org/dspace/app/rest/BitstreamRestControllerIT.java

```diff
@@ -425,5 +514,127 @@ private int getNumberOfPdfPages(byte[] content) throws IOException {
         }
     }
 
+    @Test
+    public void getBitstreamFormat() throws Exception {
+
+        getClient().perform(get("/api/core/bitstreams/" + bitstream.getID() + "/format"))
+                .andExpect(status().isUnauthorized());
 
+        getClient(getAuthToken(eperson.getEmail(), password))
+                .perform(get("/api/core/bitstreams/" + bitstream.getID() + "/format"))
+                .andExpect(status().isForbidden());
+
+        context.turnOffAuthorisationSystem();
+
+        resourcePolicy = resourcePolicyService.create(context);
+        resourcePolicy.setEPerson(eperson);
+        resourcePolicy.setAction(READ);
+        resourcePolicy.setdSpaceObject(bitstream);
+
+        context.commit();
+
+        getClient(getAuthToken(eperson.getEmail(), password))
+                .perform(get("/api/core/bitstreams/" + bitstream.getID() + "/format"))
+                .andExpect(status().isOk());
+
+        getClient(getAuthToken(admin.getEmail(), password))
+                .perform(get("/api/core/bitstreams/" + bitstream.getID() + "/format"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", matchBitstreamFormat(
+                        supportedFormat.getID(),
+                        supportedFormat.getMIMEType(),
+                        supportedFormat.getDescription(),
+                        supportedFormat.getShortDescription(),
+                        "SUPPORTED"
+                )));
+    }
+
+    @Test
+    public void updateBitstreamFormat() throws Exception {
```

Comment:
I would recommend breaking this test method up into several methods.  It seems to be doing several types of tests -- some are passing Invalid data, some are checking permissions, some are looking for success responses.  If you want to do this all in one method, then please add in inline comments to help other developers (by describing each test in human terms).  Otherwise, split this into a number of test methods (like `updateBitstreamFormatNoPermissions()`, `updateBitstreamFormatInvalidData()`, and `updateBitstreamFormat()`)

Reply:
I split the method like you suggested